config_parser.cc

```diff
@@ -152,6 +152,12 @@ bool NginxConfigParser::Parse(std::istream* config_file, NginxConfig* config) {
   config_stack.push(config);
   TokenType last_token_type = TOKEN_TYPE_START;
   TokenType token_type;
+
+  // Create a token stack to handle multiple contexts
+  // Note: I understand this can be solved with a simple counter variable.
+         
```

Comment:
It could be solved with a counter, but personally I think the stack solution is cleaner and more readable. Also prevents any weird errors from counting/incrementing incorrectly.